subsys/bluetooth/controller/multithreading_lock.c

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include "multithreading_lock.h"
+
+/* Semaphore with a single count functioning as a lock. */
+K_SEM_DEFINE(ble_controller_lock, 1, 1);
```

Comment:
It seems rather as a job for a mutex than a semaphore.

Reply:
You are correct, I don't know why a mutex wasn't considered for this.

- [ ] Use a mutex for locking